Update for RangeControl documentation

davewhitley opened the pull request:
Initial commit to update the RangeControl readme. These changes add design documentation.

## Description

This PR adds design guideline documentation in addition to the development documentation that existed previously. These guidelines are 'best practices' for the usage of the component, as well as describing the component in more detail.

## Screenshots

A preview can be seen here, although it doesn't represent what it will look like in the handbook:

[preview](https://codepen.io/anon/pen/QJMLqV?editors=1000)

## Feedback

I'd appreciate feedback on:

- Whether my markdown syntax is correct
- Design guidelines
- Double-checking a11y of the document
- Seeing if the design guidelines align with the development documentation

Thank you!

davewhitley committed:
Update for RangeControl documentation

This adds design documentation to RangeControl and also edits one sentence in the developer documentation to make it more generic.

swissspidy commented on packages/components/src/range-control/README.md:
```diff
@@ -1,17 +1,101 @@
+
+![](https://make.wordpress.org/design/files/2018/12/rangecontrol-anatomy.png)
+
+A RangeControls can contain the following elements:
```
_A RangeControl_ or just _RangeControls_, no?

chrisvanpatten commented on packages/components/src/range-control/README.md:
```diff
@@ -1,17 +1,101 @@
+
+## Table of contents
+
+1. [Design guidelines](http://#design-guidelines)
```
I think the `http://` here probably needs to be removed?

chrisvanpatten commented on packages/components/src/range-control/README.md:
```diff
@@ -1,17 +1,101 @@
+
+### Usage
+
+RangeControls reflect a range of values along a track, from which users may select a single value. They are ideal for adjusting settings such as volume, opacity, or text size.
```
This is a broader question but I wonder whether it makes more sense to refer to `RangeControls reflect` or the singular `A RangeControl reflects`. Might be worth a copy review on this.

chrisvanpatten commented on packages/components/src/range-control/README.md:
```diff
@@ -1,17 +1,101 @@
+
+##### Continuous sliders
+
+Continuous sliders allow users to select a value along a subjective range. They do not display the selected numeric value. Use them when displaying/editing the numeric value is not important, like volume.
```
Does the range component even allow this?

drw158 replied:
Not explicitly, although you can set a large max number and simulate a continuous slider. I feel like it will be a needed feature in the near future, given that it's an option in a regular [html slider](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#Setting_the_value's_granularity). Even though the component doesn't support it fully, I think the design guidelines can be somewhat aspirational in this case.

chrisvanpatten reviewed (changes requested):
A few questions but overall nice. I also noticed some of the docs here are pulled from Material Design's documentation — do we need to credit that somewhere?

davewhitley committed:
Typos and small adjustments

drw158 commented:
@chrisvanpatten 

> A few questions but overall nice. I also noticed some of the docs here are pulled from Material Design's documentation — do we need to credit that somewhere?

You're right, most of the docs are pulled from Material, even though they are mostly edited to fit WordPress. What would be the best way to credit Material? I think it would be easier if we had a separate documentation site. In terms of the component package, maybe a top level readme file in the component folder?

ajitbohra commented:
@chrisvanpatten 

Should we convert props info to the table like we have one here https://github.com/WordPress/gutenberg/tree/master/packages/components/src/button#props 

The table looks neat and easy to refer compared to vertical flowing text, though formatting on the handbook for table needs CSS tweaks.

chrisvanpatten commented:
@ajitbohra I like the more verbose version we have here as it makes it easier to deep-link to a specific prop in GitHub. Also the table makes it harder to provide in-context examples for specific props.

kjellr commented:
> You're right, most of the docs are pulled from Material, even though they are mostly edited to fit WordPress. What would be the best way to credit Material? I think it would be easier if we had a separate documentation site. In terms of the component package, maybe a top level readme file in the component folder?

I do think something like that would make sense — but maybe it could just be a note on one of these more global docs pages?

https://github.com/WordPress/gutenberg/blob/master/docs/readme.md
https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/readme.md

In any case, I don't think it should hold up this PR. It looks to me like the other issues raised here are addressed — can someone else confirm?

mapk reviewed (changes requested):
The usage of "range sliders" and "continuous sliders" appears in the **Behavior** section of the doc before I even understand the difference. When we get to the **Type** section, I only see "continuous slider" and "discrete slider" mentioned which leaves me a bit lost. 

I imagine that "range slider" is a term that includes both "continuous" and "discrete" but am unsure. Can this be made more explicit?

kjellr commented:
I agree with @mapk on that one. For the simplest fix, I think it'd be fine to move types out of "Usage" and into "Anatomy" (Or even into its own H3 maybe). Bonus points if we can include a screenshot of each, but we can also add that later on.

davewhitley committed:
Moving types higher in the doc

davewhitley committed:
Updates slider language

I removed "range slider" language so that it's not confused with the types of sliders. I also inserted "RangeControl" when referring to the whole component. "Slider" is used when casually referring to the thumb+track portion of the component.

drw158 commented:
@mapk @kjellr ready for another review

Moved Types section here https://github.com/WordPress/gutenberg/pull/12564/commits/57fa128bd4320bb2f3dcb79332d033bde0647799

Updated slider language here https://github.com/WordPress/gutenberg/pull/12564/commits/4bd4fcc033f55403e53f8964463643f46444bcc4

"Slider" refers to the thumb+track part of the component even though it's not in the official anatomy.

mapk reviewed (approved):
:shipit: Reads much better for me now. Thanks!

gziolo commented:
It’s only docs changes, we can ignore bad the fact that Travis CI is red expressing its bad mood 😃 Nice work on providing lots of useful details about this component.
